honeybee_energy/hvac/_base.py

```diff
@@ -0,0 +1,129 @@
+                to a single Room. If True, an error will be raised if the system
+                is set to more than one Room and if False, no error will be raised
+                in such cases. An example of a single room system is the IdealAirSystem
+                and an example of a multi room system is VAVWithReheat.
+        """
+        self._is_single_room = bool(is_single_room)
+        self._parent = None  # this is only used by single room systems
+        self.name = name
+    
+    @property
+    def name(self):
+        """Get or set the text string for HVAC system name."""
+        return self._name
+
+    @name.setter
+    def name(self, name):
+        self._name = valid_ep_string(name, 'HVAC system name')
+    
+    @property
+    def is_single_room(self):
+        """Get a boolean noting whether the HVAC system is assignable to only one Room.
+        """
+        return self._is_single_room
+    
+    @property
+    def schedules(self):
+        """Get an array of all the schedules assiciated with the HVAC system.
+
+        This property should be overwritten in each of the classes inheriting from
+        the HVACSystem base class since each HVAC system is likely to have it's
+        own unique places where schedules are assigned. At a minimum, this property
+        should return heating/cooling availability schedules. 
+        """
+        return ()
+
+    def duplicate(self):
+        """Get a copy of this object."""
+        return self.__copy__()
+
+    def ToString(self):
+        """Overwrite .NET ToString."""
+        return self.__repr__()
+
+    @staticmethod
+    def _check_schedule(schedule, obj_name=''):
+        """Check that an input schedule is a correct object type."""
+        assert isinstance(schedule, (ScheduleRuleset, ScheduleFixedInterval)), \
+            'Expected ScheduleRuleset or ScheduleFixedInterval for {} ' \
+            'schedule. Got {}.'.format(obj_name, type(schedule))
+
+    @staticmethod
+    def _get_schedule_from_dict(sch_dict):
+        """Get a schedule object from a schedule dictionary."""
+        if sch_dict['type'] == 'ScheduleRuleset':
+            return ScheduleRuleset.from_dict(sch_dict)
+        elif sch_dict['type'] == 'ScheduleFixedInterval':
+            return ScheduleFixedInterval.from_dict(sch_dict)
+        else:
+            raise NotImplementedError(
+                'Schedule {} is not supported.'.format(sch_dict['type']))
+
+    def __copy__(self):
+        return _HVACSystem(self.is_single_room)
+
+    def __repr__(self):
+        return 'HVACSystem:\n is single room: {}'.format(self.is_single_room)
+
+
+class _HVACEnumeration(object):
+    """Enumerates all installed HVAC classes."""
+    _HVAC_TYPES = {}
+
+    def __init__(self, import_modules=True):
+        if import_modules:
+            self._import_modules()
+
+        for clss in _HVACSystem.__subclasses__():
+            self._HVAC_TYPES[clss.__name__] = clss
```

Comment:
My first impression is that this is too much magic for creating a dictionary of HVAC system objects but you probably have a good reason for doing so.

Reply:
As I said in response to your first comment, I can take this out this `_HVACEnumeration` class in `_base` and hard code in the fact that `IdealAirSystem` is the only HVAC system that comes with `honeybee_energy`. We will still have a need for this dictionary of `HVAC_TYPES` since this dictionary is where all of the `Room.from_dict` methods look up the HVAC class in order to re-serialize various different HVAC dictionaries. But we'll just have the `honeybee_energy_hvac` extension add classes directly to this dictionary. So this extra step of searching for modules in the `hvac` subpackage can be taken out. Please let me know if you agree.

Reply:
I think this is a better approach.

Reply:
Done!